### PR TITLE
dnsmasq: Create three more optional parameters

### PR DESCRIPTION
**Pull request by meyergru**
I just found out that the default TTL for local DNS entries is just 1 second and that some nectwork clients just ask over and over for the same address again.

The "local" TTL is not yet configurable as well as two other parameters (dns_forword_max and cache_size), which is addressed by this patch.

### CODE REVIEW

File: src/www/services_dnsmasq.php [hunk 1]
```diff
@@ -371,6 +398,33 @@
                     <?= gettext('Log the results of DNS queries') ?>
                   </td>
                 </tr>
+                <tr>
+                  <td><a id="help_for_dns_forward_max" href="#" class="showhelp"><i class="fa fa-info-circle"></i></a> <?=gettext("Max number DNS queries");?></td>
+                  <td>
+                    <input name="dns_forward_max" type="text" id="dns_forward_max" size="6" placeholder="5000" <?=!empty($pconfig['dns_forward_max']) ? "value=\"{$pconfig['dns_forward_max']}\"" : "";?> />
+                    <div class="hidden" data-for="help_for_dns_forward_max">
+                      <?=gettext("The maximum number of forwarded DNS queries (default 5000)");?>
```
Comment (fichtner) [inline src/www/services_dnsmasq.php:406]: I prefer the placeholder over embedding numbers in the help text

```suggestion
                      <?=gettext("The maximum number of forwarded DNS queries.");?>
```

File: src/www/services_dnsmasq.php [hunk 2]
```diff
@@ -371,6 +398,33 @@
...
+                    </div>
+                  </td>
+                </tr>
+                <tr>
+                  <td><a id="help_for_cache_size" href="#" class="showhelp"><i class="fa fa-info-circle"></i></a> <?=gettext("Cache size");?></td>
+                  <td>
+                    <input name="cache_size" type="text" id="cache_size" size="8" placeholder="10000" <?=!empty($pconfig['cache_size']) ? "value=\"{$pconfig['cache_size']}\"" : "";?> />
+                    <div class="hidden" data-for="help_for_cache_size">
+                      <?=gettext("The number of entries in DNS cache (default 10000)");?>
```
Comment (fichtner) [inline src/www/services_dnsmasq.php:415]: Same here

File: src/www/services_dnsmasq.php [hunk 3]
```diff
@@ -371,6 +398,33 @@
...
+                    </div>
+                  </td>
+                </tr>
+                <tr>
+                  <td><a id="help_for_local_ttl" href="#" class="showhelp"><i class="fa fa-info-circle"></i></a> <?=gettext("TTL for local DNS");?></td>
+                  <td>
+                    <input name="local_ttl" type="text" id="local_ttl" size="5" placeholder="1" <?=!empty($pconfig['local_ttl']) ? "value=\"{$pconfig['local_ttl']}\"" : "";?> />
+                    <div class="hidden" data-for="help_for_local_ttl">
+                      <?=gettext("The Time-To-Live in seconds for local DNS entries, i.e. /etc/hosts or DHCP leases (default 1)");?>
```
Comment (fichtner) [inline src/www/services_dnsmasq.php:424]: Same same :)

Comment (meyergru) [reply src/www/services_dnsmasq.php]: changed

File: src/www/services_dnsmasq.php (hunk 2 above)
Comment (meyergru) [reply src/www/services_dnsmasq.php]: changed

File: src/www/services_dnsmasq.php (hunk 1 above)
Comment (meyergru) [reply src/www/services_dnsmasq.php]: changed

**Comment by fichtner** [comment]
Merged, thanks a lot!

**Comment by fichtner** [comment]
@meyergru I've gone through the man page of dnsmasq to tweak this submission: https://github.com/opnsense/core/commit/1dd42156825 https://github.com/opnsense/core/commit/23bb6c2226e

**Pull request merged**